web/packages/plugins/apps/dmt-app/src/components/context-menu/ContextMenu.tsx

```diff
@@ -1,137 +1,96 @@
-import {
-  ContextMenu,
-  MenuItem,
-  ContextMenuTrigger,
-  SubMenu,
-} from 'react-contextmenu'
+import React, { useContext } from 'react'
 import './react-contextmenu.css'
-import './custom.css'
-import styled from 'styled-components'
-
-const attributes = {
-  className: 'custom-root',
-}
-
-export interface MenuItem {
-  action?: string
-  label: string
-  menuItems?: MenuItem[]
-  icon?: any
-  data?: any
-}
-
-type DmtMenuItemsProps = {
-  id: string
-  onClick: Function
-  menuItems: MenuItem[]
-}
-
-const renderMenuItems = (props: DmtMenuItemsProps) => {
-  const { id, menuItems, onClick } = props
-  return menuItems.map((menuItem: MenuItem, index: number) => {
-    const key = `menuitem-${id}-${index}`
-    if (Object.keys(menuItem).length === 0) {
-      return <MenuItem divider />
-    } else if (menuItem.menuItems) {
-      return (
-        <DmtSubMenu
-          id={id}
-          key={key}
-          onClick={onClick}
-          menuItems={menuItem.menuItems}
-          label={menuItem.label}
-        />
-      )
-    } else {
-      return (
-        <DmtMenuItem key={key} id={id} onClick={onClick} menuItem={menuItem} />
-      )
-    }
-  })
-}
-
-type DmtSubMenuProps = {
-  id: string
-  label: string
-  menuItems: MenuItem[]
-  onClick: Function
-}
-
-const UnClickable = ({ children }: any) => (
-  <span onClick={(e: any) => e.stopPropagation()}>{children}</span>
-)
-
-const DmtSubMenu = (props: DmtSubMenuProps) => {
-  const { id, label, onClick, menuItems } = props
-
-  const menuItemsComponents = renderMenuItems({
-    id,
-    onClick,
-    menuItems,
-  })
-
-  return (
-    <UnClickable>
-      <SubMenu title={<span>{label}</span>}>{menuItemsComponents}</SubMenu>
-    </UnClickable>
-  )
+import { ContextMenu, ContextMenuTrigger, MenuItem } from 'react-contextmenu'
+import { AuthContext, BlueprintEnum, DmssAPI, TreeNode } from '@dmt/common'
+
+// @ts-ignore
+import { NotificationManager } from 'react-notifications'
+function sortMenuItems(menuItems: JSX.Element[]) {}
+
+function createMenuItems(
+  node: TreeNode,
+  dmssAPI: DmssAPI,
+  removeNode: Function
+): JSX.Element[] {
+  let menuItems = []
+
+  const DeleteAction = (node: TreeNode) => {
+    dmssAPI
+      .removeByPath(node.pathFromRootPackage(), node.dataSource)
+      .then(() => {
+        removeNode(node)
+        NotificationManager.success('Deleted')
+      })
+      .catch((error: Error) => {
+        console.error(error)
+        NotificationManager.error(
+          JSON.stringify(error.message),
+          'Failed to delete'
+        )
+      })
+  }
+  const ViewAction = (node: TreeNode) => {
+    // @ts-ignore
+    window.open(`dmt/view/${node.nodeId}`, '_blank').focus()
+  }
+
+  // dataSources get a "new root package"
+  if (node.type === 'dataSource') {
+    menuItems.push(
+      <MenuItem key={'new-root-package'} onClick={() => {}}>
+        New package
+      </MenuItem>
+    )
+  }
+
+  // Packages get a "new folder"
```

Comment:
maybe we also new "new blueprint" and "new entity"?

Reply:
issues created